bokeh/models/ranges.py

```diff
@@ -90,6 +100,19 @@ class DataRange1d(DataRange):
     automatically computed end value.
     """)
 
+    bound_lower = Float(help="""
+    The minimum value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data. If not provided, will
+    default to the min of the data.
+    """)
```

Comment:
I think this is not quite adequate, I would suggest `None` means no bound, `"auto"` means default to the min of the data, and explicit values are explicit bounds. So this would be `Either(Auto, Float)`